### ISSUE TITLE
Add Aurelia support

### ISSUE THREAD

**Issue by enriqueMartinezInter** [stackblitz/core#19]
Please add aurelia support.

**Comment by EricSimons** [comment]
@enriqueMartinezInter would love to add this! Does Aurelia have a CLI/etc that's based on webpack?

**Comment by enriqueMartinezInter** [comment]
I don't think it´s based on webpack no. Not at the moment at least.

**Comment by enriqueMartinezInter** [comment]
Actually I just got confirmation that the Aurelia CLI will support Webpack pretty soon :D

**Comment by avrahamcool** [comment]
aurelia cli supports webpack for a while now..
what do we have to do for this to work?
i'll gladly help on the matter.

**Comment by EricSimons** [comment]
@eisenbergeffect we'd love to get a starter template set up for Aurelia on stackblitz! We also have a [new SDK in beta](https://gist.github.com/EricSimons/72017ec7ba068a5b492ee39f9e3a7f32) that makes it super easy to set up for docs/etc, if desired :)

What's a good way to get in touch with y'all about getting the ball rolling on this?

**Comment by EisenbergEffect** [comment]
Hey @EricSimons ! Thanks for reaching out. Feel free to email me at rob at bluespire dot com or continue the conversation here on this issue. Whatever works for you is fine with me. Let me know what you've got in mind and how we can assist you in getting Aurelia supported. Thanks!

**Comment by adriatic** [comment]
@EricSimons 

If it was not the case that the gentlemen who has my utmost respect (@EisenbergEffect of course) has already registered his interest in Stackblitz, I would be already all over you now.

Rob, please let me know (using our private Gitter channel) if there is a way that I can play any role in making Aurelia supported on Stackblitz - including testing, writing samples, docs, ...

**Comment by adriatic** [comment]
I contacted @EricSimons after I discussed the Aurelia support on SB with @EisenbergEffect, to ensure that I am not stepping into @EisenbergEffect's plans. Eric and I had a good discussion on about the plan to use SB as a component of the AUCS portal, allowing the AUCS Guide's samples to run in the context of each Guide - precisely the scenario SB was designed for. As a part of this plan, we (AUCS team) will create the needed support (Aurelia Template) for SB.

The nature of our relationship with SB is that Eric will help us do the integration - not the other way around.

**Comment by Alexander-Taran** [comment]
@EricSimons  can I read somwhere on how do you do the magic? I parsed the template repos. And got the basic understanding that whatever is created in "project" inside editor is mapped to src folder of 'create-react-app' for example..
so if i'll create kinda same setup with Aurelia-CLI.. how can we try it?

I'd rather see Aurelia setup @ stackblitz soner than later. 
I'm the guy answering a lot of newbie questions in aurelia channels every day and I'd love to have beautiful answers packed with stackblitz being part of FAQ and other guides.
So I've got enough motivation to make it happen yesterday.

@EisenbergEffect said that'd be awesome If I could jump on working on this.
So here I am. Jumping on (-:

**Comment by EisenbergEffect** [comment]
@EricSimons We've had some conversation on our side and @Alexander-Taran will work with you from the Aurelia side to help put together what is needed for general Aurelia integration with StackBlitz. Once that is looking good, we'll loop in @adriatic to do some integration with his project and I'll also work on integration with the Aurelia docs. @EricSimons I think @Alexander-Taran has some open questions for you that would help him take his next steps. Thanks for your patience and for working with us!

**Comment by adriatic** [comment]
This seems like a quick convergence that makes all parties happy as well

**Comment by EricSimons** [comment]
@EisenbergEffect @Alexander-Taran sorry about my delay here! Super excited to get aurelia support rolling :)

@Alexander-Taran do you think it'd be possible to take one of our existing templates and make Aurelia work in it? I could then take any custom config/etc files you created for it and store them under the hood for the actual template. If you run into any roadblocks with that approach I can also tweak our bundler too. Lmk!

**Comment by Alexander-Taran** [comment]
@EricSimons I guess that CLI based setups are prefered in case of import export?
I will scaffold a CLI based structure and a 'minimal' webpack structrure and put them in separate repos. As starting points. I'll try to stay close to the existing templates.

Out of the box I cant make existing templates work. We  need aurelia-webpack plugin to enlighten webpack on Aurelias conventions (-:

I'll get back to you as soon as I have them

**Comment by Alexander-Taran** [comment]
@EricSimons first stab at a minimal template
https://github.com/aurelia-contrib/stackblitz-minimal-webpack-ts-template

without your custom  {BLITZ_RESOURCES} or {BLITZ_TEMPLATE} yet.

Questions:
Do I understand it right that resources will contain webpack bundle produced?

And about the template. Aurelia convention is to put aurelia-app attribute on body.  
After the bootstrap of aurelia will happen the contents inside would be removed from dom.
What will happen if we add {BLITZ_TEMPLATE} inside? What part does it play?
```
  <body aurelia-app="main">
    <script src="/dist/bundle.js"></script>
  </body>
```

**Comment by avrahamcool** [comment]
any news on this?

**Comment by Alexander-Taran** [comment]
@avrahamcool waiting for static resource support

**Comment by EricSimons** [comment]
We're working on the new bundler & dev server right now actually! Should be fully online by EOM w/ static assets support/etc :)

**Comment by Alexander-Taran** [comment]
@avrahamcool meanwhile, you can play with forks of this one https://stackblitz.com/edit/aurelia-base?file=app.js

**Comment by adriatic** [comment]
@Alexander-Taran can you share some details what would this "Stackblitz Aurelia" work once the new bundler and dev server will be finished?

**Comment by djedi** [comment]
Just saw a demo of StackBlitz today. Watching this thread! +1

**Comment by VagyokC4** [comment]
Any updates on this?

**Comment by Alexander-Taran** [comment]
@VagyokC4 there is codesandbox https://codesandbox.io/s/j3mjmnwrnw
sorry @EricSimons 
We should continue our efforts.

**Comment by EricSimons** [comment]
@Alexander-Taran hey- would you still be interested in helping make this work? Lmk!

**Comment by glyad** [comment]
Any updates on this?

**Comment by fvsch** [comment]
In 2021 we introduced support for running Node.js-based projects using [our WebContainers runtime](https://developer.stackblitz.com/guides/user-guide/available-environments).

This should allow running most frameworks that depend on Node.js, npm packages, JavaScript and/or WebAssembly.

I did not find an example repository with a working Aurelia project to try in WebContainers, but given the above information I suspect it should work.

If there are specific issues with running Aurelia with WebContainers, feel free to open a new issue with those specific issues.
